akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedBehavior.scala

```diff
@@ -37,13 +37,11 @@ private[akka] object EventsourcedBehavior {
     // format: ON
   }
 
-  def withMdc(setup: EventsourcedSetup[_, _, _], phaseName: String)(b: Behavior[InternalProtocol]): Behavior[InternalProtocol] = {
-    val mdc = Map(
+  def withMdc(setup: EventsourcedSetup[_, _, _], phaseName: String)(b: Behavior[InternalProtocol]): Behavior[InternalProtocol] =
+    Behaviors.withMdc(Map(
```

Comment:
I think we also wanted to change EventsourcedRunning to not use this `withMdc` when switching between RunningCmds and PersistingEvents all the time. Instead use the `log.withMdc` when that one is logging.

Reply:
I think that was the original reason for creating https://github.com/akka/akka/issues/24684

Reply:
yes, the purpose of the in-line one being requested was to be able to avoid the extensive wrapping we'd have to do otherwise -- the phase param should be handled in the phases now so when persistnig events the mdc would contain that, and when handling commands also that would be reflected in the phase mdc